src/benchmarks/micro/corefx/System.Text.Json/Utf8JsonWriter/Perf.Large.cs

```diff
@@ -0,0 +1,138 @@
+
+using System.Buffers;
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+
+namespace System.Text.Json
+{
+    [BenchmarkCategory(Categories.CoreFX, Categories.JSON)]
+    public class Perf_Large
+    {
+        private static readonly byte[] ExtraArray = Encoding.UTF8.GetBytes("ExtraArray");
+        private static readonly byte[] First = Encoding.UTF8.GetBytes("first");
+        private static readonly byte[] Last = Encoding.UTF8.GetBytes("last");
+        private static readonly byte[] Age = Encoding.UTF8.GetBytes("age");
+        private static readonly byte[] PhoneNumbers = Encoding.UTF8.GetBytes("phoneNumbers");
+        private static readonly byte[] Address = Encoding.UTF8.GetBytes("address");
+        private static readonly byte[] Street = Encoding.UTF8.GetBytes("street");
+        private static readonly byte[] City = Encoding.UTF8.GetBytes("city");
+        private static readonly byte[] Zip = Encoding.UTF8.GetBytes("zip");
+
+        private const int DataSize = 100_000;
+
+        private ArrayBufferWriter<byte> _arrayBufferWriter;
+        private PooledBufferWriter<byte> _pooledBufferWriter;
+
+        private string[] _propertNames;
+        private byte[][] _propertNamesUtf8;
+        private int[] _numberArrayValues;
+        private string[] _stringArrayValues;
+
+        [Params(true, false)]
+        public bool Formatted;
+
+        [Params(true, false)]
+        public bool SkipValidation;
+
+        [Params(true, false)]
+        public bool Pool;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            _arrayBufferWriter = new ArrayBufferWriter<byte>();
+            _pooledBufferWriter = new PooledBufferWriter<byte>();
+
+            var random = new Random(42);
+
+            _propertNames = new string[DataSize];
+            _propertNamesUtf8 = new byte[DataSize][];
+            _numberArrayValues = new int[DataSize];
+            _stringArrayValues = new string[DataSize];
+
+            for (int i = 0; i < DataSize; i++)
+            {
+                _propertNames[i] = "abcde" + i.ToString();
+                _propertNamesUtf8[i] = Encoding.UTF8.GetBytes(_propertNames[i]);
+                int value = random.Next(-10000, 10000);
+                _numberArrayValues[i] = value;
+                _stringArrayValues[i] = value.ToString();
+            }
+        }
+
+        [Benchmark]
+        public void WriteLargeUtf8()
+        {
+            IBufferWriter<byte> output = GetOutput();
+            var state = new JsonWriterState(options: new JsonWriterOptions { Indented = Formatted, SkipValidation = SkipValidation });
+            var json = new Utf8JsonWriter(output, state);
+
+            json.WriteStartObject();
+            for (int i = 0; i < DataSize; i++)
+            {
+                json.WriteStartArray(_propertNamesUtf8[i]);
+
+                int number = _numberArrayValues[i];
+
+                if (number > 0)
```

Comment:
Generally, a benchmark should be testing a single path. What's the goal of testing `WriteStringValue`/`WriteNumberValue` in concert like this?

Reply:
This is to avoid having the JIT over-predict the method calls. Generally, users won't just keep calling a single method and their payload will contain diverse data types. If you have a strong opinion on this, I can change it, and just write both without the check.

Reply:
I would write both. The JIT isn't going to go out of its way here to elide this call due to its complexity.